### ISSUE TITLE
code for generating input from real world imagery

### ISSUE THREAD

**Issue by waldstein94** [yasaminjafarian/HDNet_TikTok#4]
Hi yasamin, thanks to your great work, and I've really enjoyed it! 

Here is my question. 
I want to run your code on my images taken with my phone, but there is no document or requirement to make UV map.
Do I just use the output uv map of DensePose for the input of test code, or is there something other process I need to do?

Thank you for your answer in advance :)

**Comment by yasaminjafarian** [comment]
Hi. Thanks a lot for your message. You just need to use the output IUV map you get from DensePose for the input to the test code. No processing is needed. An example of that is in the [test folder](https://github.com/yasaminjafarian/HDNet_TikTok/tree/main/test_data). For each image, you need to have image (name_img.png), human segmentation (name_mask.png) and depnsepose output (name_dp.png). Just note that these images should be cropped to the size 256x256. 
For DensePose you can simply install it by the [docker image](https://github.com/facebookresearch/DensePose/blob/master/INSTALL.md#docker-image) they provide and run it on any image you want to feed to this code.

**Comment by waldstein94** [comment]
Thank you for the answer. I'm gonna check them out!

And also there were several questions raised after reading your paper.
Sorry for too many questions in advance.
 
1) HDNet is trained with the loss composed of four terms mixed in supervised and self-supervised learning manner. Did you optimize four terms at once or using other train scheme like training it first in supervised manner, and then in the self-supervised manner?

2) Why do HDNet need UV map as input? I think I probably understood what your concept is and what you expected from using UV map as input, which is indicated as equation (4) in your paper. However, I'm not sure if the network does really learn about some relations between UV, 3D point and Image. Have you trained the network without UV map?

3) You used 340 subjects with 3D ground truth, and how did you choose that number? Could there have been an overfitting problem when using less than 340 subject? Also, which the criteria did you use when selecting the subject? There are several attributes such as pose, race, clothes, etc.. 

4) Does your network consider the perspective effect when estimating depth? For example, the last image in figure (7) in the paper seems to be taken upper view, and the result also seems pretty accurate. Without any specific algorithm for the perspective effect, the network appears to learn it very well.  

5) Finally, this is the last question. You adopted self-supervised learning manner with supervised learning manner. Then, have you tried to train the network with only self-supervised loss term? If you did, could you let me know how the result was.

**Comment by yasaminjafarian** [comment]
No problem. Please find my answers here:
1. The supervised losses (Ln and Lz) has been pretrained on the ground truth data and then all together (Ln, Lz, Ls, Lw) are retrained and optimized with the ground truth data and unlabeled TikTok data together.
2. It sure doesn't need the UV map as input. I just trained it in such configuration to let the network know about the location of each body part. The test code without the UV as input will be added soon.
3. We just bought almost all the available data last year from RenderPeople. We didn't consider the variety of that. Note that the available data in RenderPeople is much bigger now.
4. It does not but we saw as you mentioned for the oblique view (e.g. Vlastic et. al. dataset) it can generalize well. 
5. We didn't try with just self-supervised loss.

**Comment by waldstein94** [comment]
Thanks a lot!! Your answers are very helpful to me. 
Have a good day :)
